test/studies/shootout/nbody/bradc/nbody-blc.chpl

```diff
@@ -116,14 +114,10 @@ proc energy() {
   var e = 0.0;
 
   for i in 0..<numBodies {
-    const b1 = bodies[i];
```

Comment:
Getting rid fo the ref and this copy seem like they should be fine. In the ref case it avoids an extra index operation, but those were pretty optimized for DR arrays so I'm not sure how much benefit I'd expect (or if the backend can determine it's the same index and do a similar optimization automatically)

In this case, we're adding more indexing, but removing a copy. It may be worth playing with these changes in isolation. I might start by making this ref and seeing if avoiding the copy matters, and then remove the refs to see if the extra indexing hurts.

Reply:
This routine is only called twice during the whole execution, so it's not a big time sync no matter how it's written (compared to the 50000000 times that advance is called).  So I haven't done a lot to optimize this case (e.g., no param loops) and switched to the indexing primarily for consistency in style with advance() rather than anything deep.

Reply:
Ah, I missed that it's not called much.